core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1928,9 +1928,11 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
 
   private def getInterBrokerListenerNameAndSecurityProtocol: (ListenerName, SecurityProtocol) = {
     Option(getString(KafkaConfig.InterBrokerListenerNameProp)) match {
-      case Some(_) if originals.containsKey(KafkaConfig.InterBrokerSecurityProtocolProp) =>
-        throw new ConfigException(s"Only one of ${KafkaConfig.InterBrokerListenerNameProp} and " +
-          s"${KafkaConfig.InterBrokerSecurityProtocolProp} should be set.")
+      case Some(_) if originals.containsKey(KafkaConfig.InterBrokerSecurityProtocolProp) &&
+        getString(KafkaConfig.InterBrokerSecurityProtocolProp).nonEmpty =>
+          throw new ConfigException(s"Only one of ${KafkaConfig.InterBrokerListenerNameProp} : " +
+            s"${getString(KafkaConfig.InterBrokerListenerNameProp)} and ${KafkaConfig.InterBrokerSecurityProtocolProp} : " +
```

Comment:
Bikeshedding comment:

Would it be possible to change line 1931 from `Some(_)` to `Some(foo)` and then use the variable on line 1934? Just wondering if that improves readability?

Reply:
Done~  Thanks for your review.